pkg/controller/bastion/resourcesdefine.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bastion
+
+import (
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/security/rules"
+)
+
+// IngressAllowSSH ingress allow ssh
+func IngressAllowSSH(opt *Options, secGroupID string) rules.CreateOpts {
+	return rules.CreateOpts{
+		Direction:      "ingress",
+		Description:    ingressAllowSSHResourceName(opt.BastionInstanceName),
+		PortRangeMin:   1,
+		EtherType:      rules.EtherType4,
+		PortRangeMax:   65535,
+		Protocol:       "tcp",
+		SecGroupID:     secGroupID,
+		RemoteIPPrefix: "0.0.0.0/0",
+	}
+}
+
+// EgressAllowSSHToWorker egress allow ssh to worker
+func EgressAllowSSHToWorker(opt *Options, secGroupID string, remoteGroupID string) rules.CreateOpts {
+	return rules.CreateOpts{
+		Direction:     "egress",
+		Description:   egressAllowOnlyResourceName(opt.BastionInstanceName),
+		PortRangeMin:  1,
+		EtherType:     rules.EtherType4,
+		PortRangeMax:  65535,
+		Protocol:      "tcp",
+		SecGroupID:    secGroupID,
+		RemoteGroupID: remoteGroupID,
+	}
+}
```

Comment:
It seems `egress deny all` created by default when I create security group
![image](https://user-images.githubusercontent.com/42234376/149272222-c799f4ad-a981-4ec3-ae4a-39d789ab9e9d.png)

Reply:
in this case, please add a comment in this file like: `nsgEgressDenyAll` for the bastion is not required on openstack as all egress is denied by default.

Reply:
however, when I think about it again, it doesn't hurt if we would add it to the code, it would make it more explicit, even though it might be there by default. So let's add it to the code and ignore my previous reply.

Reply:
done b21e488b

Reply:
@tedteng have you tested that it works as expected? Because this is what I found in the documentation:

> Security rules are always permissive; you can’t create rules that deny access. 

This means that your change would not work as expected. Instead you would need to make sure that there are no other rules for the bastion security group that we didn't create, or not?

Reply:
The deny rules `EgressDenyAll` and `EgressAllowSSHToWorker` seems not work as expected, 

still able to `ping 8.8.8.8` as `egress IPv4 deny all Any Any` 

bastion node(10.250.3.x) able to ping `shoot` node(10.250.3.124) after removed rule `EgressAllowSSHToWorker` . 
Search the IP `10.250.3.2xx` and `10.250.3.1xx` from the existing Servers list, unable to ping from the bastion node.

remove `EgressDenyAll` and `EgressAllowSSHToWorker` 
efe4f75e